### ISSUE TITLE
Performance: Why isn't the router started synchronously on startup?

### ISSUE THREAD

**Issue by bradvogel** [iron-meteor/iron-router#1304]
Why does iron route use `Meteor.defer()` here: https://github.com/iron-meteor/iron-router/blob/86bab2112e44c2e0649006b4ce2f73927e5b0da3/lib/router.js#L44

Rendering performance is critical in my application, and this Meteor.defer() costs me an extra 200ms or so (since my app is loaded in parallel with a few other scripts competing for CPU, and the browser just waits a long time to fire that setTimeout).

**Comment by cmather** [comment]
It's to give the chance to configure the router by calling
Router.configure, including the ability to not automatically render.
Perhaps we could take an option to render immediately. If you have a
solution that would fix your issue, happy to take a PR.

**Comment by bradvogel** [comment]
Ok, using the following code:

```
Router.configure({
  autoStart: false
});

Meteor.startup(function() {
  // Non-deferred startup. See https://github.com/iron-meteor/iron-router/issues/1304.
  Router.start();
});
```

Saves about 200ms off my app startup time. I can file a PR to make this an option. What would you call it though? Perhaps something like:

```
Router.configure({
  fastStart: true
});
```

But what is `autoStart` for anyways? I don't see any docs on it.

**Comment by bradvogel** [comment]
Ping @cmather

**Comment by cmather** [comment]
Hey brad sorry I have to look at the code again to see precisely. Little swamped at moment but will try to look tonight.

**Comment by bradvogel** [comment]
bump @cmather :)

**Comment by cmather** [comment]
You can learn more about this function by going to `lib/router.js` and check out lines 43 - 49. The defer line is to give your app code time to execute before automatically starting the router. Then to see what the start method does, look in `lib/router_client.js` down on line 171. It's primary purpose is to set up a computation to respond to url changes. On the server in `lib/router_server.js` the `start` method instead connects the router automatically to the Web connect handlers.

So autostart on the client/browser is what starts responding to url changes and dispatching into your route functions/controllers. If you don't start the router it won't dispatch.
